services/datamanager/datasync/sync.go

```diff
@@ -274,32 +276,24 @@ func exponentialRetry(cancelCtx context.Context, fn func(cancelCtx context.Conte
 
 	// First call failed, so begin exponentialRetry with a factor of RetryExponentialFactor
 	nextWait := time.Millisecond * time.Duration(InitialWaitTimeMillis.Load())
-	ticker := time.NewTicker(nextWait)
 	for {
 		if err := cancelCtx.Err(); err != nil {
 			return err
 		}
-		select {
-		// If cancelled, return nil.
-		case <-cancelCtx.Done():
-			ticker.Stop()
-			return cancelCtx.Err()
-			// Otherwise, try again after nextWait.
-		case <-ticker.C:
-			if err := fn(cancelCtx); err != nil {
-				// If error, retry with a new nextWait.
-				ticker.Stop()
-				nextWait = getNextWait(nextWait)
-				ticker = time.NewTicker(nextWait)
-				continue
-			}
-			// If no error, return.
-			ticker.Stop()
-			return nil
+		if err := fn(cancelCtx); err != nil {
```

Comment:
I think we should prefer ticker to sleep. Sleep does not take into account execution time, so if the rest of the code takes 10 seconds, `getNextWait()` will actually occur every nextWait + 10 seconds instead of every nextWait seconds. Ticker also adjusts the intervals or drops ticks to make up for slow receivers, instead of sleep which is more like an infinite loop. Ticker is also concurrency safe since it uses a channel

Reply:
I'm probably misunderstanding this but based on reading the docs and your description I still don't quite understand why we want a ticker. It would seem that tickers are better if something is happening at a specific cadence (i.e. check if we're offline every 30 seconds) and sleep is used if we want to pause execution time by some changing amount. Therefore, by changing `nextWait` we're specifically introducing an anti-pattern and we _shouldn't_ be using these tickers and should use sleep.
That is, we're basically just creating a new ticker with a new cadence every single time we run and this is no longer really functionally a ticker at all, but instead is a sleep masquerading as a ticker.

That is to say: If the execution time is 10 seconds, then it's my understanding that we _want_ to wait 10 seconds + `nextWait` because _we don't have a set cadence_. 

But again, I'm no expert and if I'm still not making sense, then I'll be happy to change!

Reply:
That makes a lot of sense re: using sleep for pausing execution for some changing amount! Just making sure that sleep respects canceling a context (i.e. `if err := fn(cancelCtx); err != nil {` results in a clean termination when shutting down viam server)

Reply:
+1, could you test ctrl-C shutting down the viam-server with an exponential retry locally? (can reduce it to run more frequently to get it to hit immediately)